src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -108,55 +116,59 @@ protected AbstractEntityAIStructure(@NotNull final J job)
     {
         super(job);
         this.registerTargets(
-          /**
-          * Select the appropriate State to do next.
-          */
-          new AITarget(START_BUILDING, this::startBuilding),
-          /**
-           * Check if we have to build something.
-           */
-          new AITarget(IDLE, this::isThereAStructureToBuild, () -> AIState.START_BUILDING),
-          /**
-           * Clear out the building area.
-           */
-          new AITarget(CLEAR_STEP, generateStructureGenerator(this::clearStep, AIState.BUILDING_STEP)),
-          /**
-           * Build the structure and foundation of the building.
-           */
-          new AITarget(BUILDING_STEP, generateStructureGenerator(this::structureStep, AIState.DECORATION_STEP)),
-          /**
-           * Decorate the AbstractBuilding with torches etc.
-           */
-          new AITarget(DECORATION_STEP, generateStructureGenerator(this::decorationStep, AIState.COMPLETE_BUILD)),
-          /**
-           * Spawn entities on the structure.
-           */
-          new AITarget(SPAWN_STEP, () -> AIState.IDLE),
-          /**
-           * Finalize the building and give back control to the ai.
-           */
-          new AITarget(COMPLETE_BUILD, this::completeBuild)
+
+                /**
+                 * Check if tasks should be executed.
+                 */
+                new AITarget(this::checkIfCanceled, IDLE),
+                /**
+                 * Select the appropriate State to do next.
+                 */
+                new AITarget(START_BUILDING, this::startBuilding),
+                /**
+                 * Check if we have to build something.
+                 */
+                new AITarget(IDLE, this::isThereAStructureToBuild, () -> AIState.START_BUILDING),
+                /**
+                 * Clear out the building area.
+                 */
+                new AITarget(CLEAR_STEP, generateStructureGenerator(this::clearStep, AIState.BUILDING_STEP)),
+                /**
+                 * Build the structure and foundation of the building.
+                 */
+                new AITarget(BUILDING_STEP, generateStructureGenerator(this::structureStep, AIState.SPAWN_STEP)),
+                /**
+                 * Spawn entities on the structure.
+                 */
+                new AITarget(SPAWN_STEP, generateStructureGenerator(this::spawnEntity, AIState.DECORATION_STEP)),
+                /**
+                 * Decorate the AbstractBuilding with torches etc.
+                 */
+                new AITarget(DECORATION_STEP, generateStructureGenerator(this::decorationStep, AIState.COMPLETE_BUILD)),
+                /**
+                 * Finalize the building and give back control to the ai.
+                 */
+                new AITarget(COMPLETE_BUILD, this::completeBuild)
         );
     }
 
     private AIState completeBuild()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 35 to the 20 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)